cardano-crypto-class/src/Cardano/Crypto/EllipticCurve/BLS12_381.hs

```diff
@@ -0,0 +1,48 @@
+{-#LANGUAGE ScopedTypeVariables #-}
+{-#LANGUAGE TypeApplications #-}
+{-#LANGUAGE FlexibleContexts #-}
+module Cardano.Crypto.EllipticCurve.BLS12_381
+(
+  -- * Types
+    P
+  , P1
+  , P2
+  , PT
+  , Curve1
+  , Curve2
+  , BLSTError (..)
+
+  -- * BLS Class
+  , BLS
+  , BLS_P
+  , BLS_Curve
+
+  -- * Point / Group operations
+  -- | These work on both curves, and take phantom parameters of type 'Curve1'
+  -- or 'Curve2' to select one of the two provided elliptic curves.
+  , inGroup
+  , addOrDouble
+  , mult
+  , cneg
+  , neg
+  , compress
+  , serialize
+  , uncompress
+  , deserialize
+  , hash
+  , generator
+  , isInf
```

Comment:
I don't know what is the Haskell convention here. To be honest, I'm no worried of enforcing qualified imports here. I would keep them short and require qualified imports. @tdammers, do you have a preference here?

Reply:
We have a words that are used very often, even in this repo: `hash` and `(un)compress`, `(de)serialize`

It is better to disambiguate (even for the purpose of grepping the repo)

Even C functions are prefixed (obviously reasons why don't apply to Haskell), but I  think it would match up nicely.

I strongly suggest a `bls` prefix

Reply:
Ok, will do 👍